client/src/components/AddPlace.js

```diff
@@ -0,0 +1,512 @@
+		},
+		{
+		  name: "New Jersey",
+		  abbreviation: "NJ"
+		},
+		{
+		  name: "New Mexico",
+		  abbreviation: "NM"
+		},
+		{
+		  name: "New York",
+		  abbreviation: "NY"
+		},
+		{
+		  name: "North Carolina",
+		  abbreviation: "NC"
+		},
+		{
+		  name: "North Dakota",
+		  abbreviation: "ND"
+		},
+		{
+		  name: "Ohio",
+		  abbreviation: "OH"
+		},
+		{
+		  name: "Oklahoma",
+		  abbreviation: "OK"
+		},
+		{
+		  name: "Oregon",
+		  abbreviation: "OR"
+		},
+		{
+		  name: "Pennsylvania",
+		  abbreviation: "PA"
+		},
+		{
+		  name: "Rhode Island",
+		  abbreviation: "RI"
+		},
+		{
+		  name: "South Carolina",
+		  abbreviation: "SC"
+		},
+		{
+		  name: "South Dakota",
+		  abbreviation: "SD"
+		},
+		{
+		  name: "Tennessee",
+		  abbreviation: "TN"
+		},
+		{
+		  name: "Texas",
+		  abbreviation: "TX"
+		},
+		{
+		  name: "Utah",
+		  abbreviation: "UT"
+		},
+		{
+		  name: "Vermont",
+		  abbreviation: "VT"
+		},
+		{
+		  name: "Virginia",
+		  abbreviation: "VA"
+		},
+		{
+		  name: "Washington",
+		  abbreviation: "WA"
+		},
+		{
+		  name: "West Virginia",
+		  abbreviation: "WV"
+		},
+		{
+		  name: "Wisconsin",
+		  abbreviation: "WI"
+		},
+		{
+		  name: "Wyoming",
+		  abbreviation: "WY"
+		}
+	];
+
+	const CityStates = {
+			Select: [],
+			Massachusetts: ["Boston"],
+			Colorado: ["Denver", "Boulder"],
+			California: ["Los Angeles", "San Francisco", "San Diego", "Fresno"]
+		}
+
+	const hours = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+	const renderedStates = States.map(state => {
+	    return <option key={state.abbreviation}>{state.abbreviation}</option>;
+	  });
+	const renderedHours = hours.map((hour, index) => {
+	    return <option key={index}>{hour}</option>;
+	  });
+	const renderedStateDropdown = States.map((loc, index) => {
+		return <option key={index}>{loc.name}</option>
+	})
+
+	const [addPlace, { loading, error }] = useMutation(ADD_PLACE, {
+													onError: (error) => console.log(error.message)
+	});
+
+	const initialState = {
+	    businessName: "",
+	    category: "",
+	    website: "",
+	    phone: "",
+	    address: "",
+	    address2: "",
+	    city: "",
+	    state: "Select",
+	    locationId: 1234,
+	    zip: "",
+	    parking: [],
+	    hours: {
+	    	monday1: null,
+	    	monday2: null,
+	    	tuesday1: null,	    		    	
+	    	tuesday2: null,
+	    	wednesday1: null,
+	    	wednesday2: null,
+	    	thursday1: null,
+	    	thursday2: null,
+	    	friday1: null,
+	    	friday2: null,
+	    	saturday1: null,
+	    	saturday2: null,
+	    	sunday1: null,
+	    	sunday2: null,
+	    },
+	    photos: "",
+	    errorMessage: ""
+	  };
+
+	const [data, updateData] = useState(initialState);
+
+    const renderedCityDropdown = CityStates[data.state].map((city, index) => {
+		return <option key={index}>{city}</option>
+	})
+
+	// Finish the submit request
+	const handleSubmit = (event) => {
+		event.preventDefault()
+		if (!data.businessName || !data.category || data.address) {
```

Comment:
What is the logic behind this if statement?

Reply:
it's supposed to prevent the user from hitting submit before they fill out the required fields...however that IF statement looks all messed up. I will correct that.

Reply:
Ok thats what i figured. Update to that would be good for temporary solution but dont worry too much, I plan on adding server side validation which will replace that eventually I believe.

Reply:
Ok that's what I figured we would do. I just wanted to put some basic validation in for now. thanks!

Reply:
If you use a form package, you can add on validators that prevent the form from submitting.  You can also build an input component that automatically displays validation errors.  I recommend checking out Formik/Yup.js if you have time.  Redux-Form and Re-Validate are pretty popular too.

Reply:
I have used formik a little and it does make it pretty simple to handle all those extras. Up to you @pjconnolly12 if you want to switch it up to one of those. It looks like the basic functionality is there now and I'm sure we can get it all going home-rolled too.

Reply:
I just updated the validation so it works correctly. I just stuck with the basics for now but I checked out Formik and it does look cool. I am going to get the design going now and might switch this to Formik when I do. but this Pull should be good now. Thanks!